distribution/src/bin/elasticsearch

```diff
@@ -20,6 +20,34 @@ if [ -z "$ES_TMPDIR" ]; then
   ES_TMPDIR=`"$JAVA" -cp "$ES_CLASSPATH" org.elasticsearch.tools.launchers.TempDirectory`
 fi
 
+# get keystore password before setting java options to avoid
+# conflicting GC configurations for the keystore tools
+unset KEYSTORE_PASSWORD
+KEYSTORE_PASSWORD=
+# If keystore exists and is encrypted, prompt for a password
+if [ -f "$ES_PATH_CONF"/elasticsearch.keystore ] \
+    && ! echo $* | grep -E '(^-h |-h$| -h |--help$|--help )' >/dev/null 2>&1 \
+    && ! echo $* | grep -E '(^-V |-V$| -V |--version$|--version )' >/dev/null 2>&1 \
+    && ! "`dirname "$0"`"/elasticsearch-keystore list <&- >/dev/null 2>&1
```

Comment:
To make sure I understand the suggestion: do you mean a subcommand for `elasticsearch-keystore` such that we can run `elasticsearch-keystore has-passphrase` here, rather than running `list` with standard input closed?

Reply:
Yes exactly. Or alternatively, something like an "info" command that would emit properties about the keystore.